_components/form-controls/10-file-input.md

```diff
@@ -0,0 +1,54 @@
+---
+title: File input
+parent: Form controls
+order: 10
+lead: File inputs allow users to attach a file
+---
+
+{% include code/preview.html component="file-input" %}
+{% include code/accordion.html component="file-input" %}
+
+<div class="usa-accordion usa-accordion--bordered site-accordion-docs">
+  <button class="usa-button-unstyled usa-accordion__button"
+      aria-expanded="true" aria-controls="file-input-docs">
+    Guidance
+  </button>
+  <div id="file-input-docs" aria-hidden="false" class="usa-accordion__content site-component-usage">
+
+    <h4>When to use the file input</h4>
+    <ul class="usa-content-list">
+      <li><strong>Documents are required.</strong> You should only ask users to upload files when it is critical to service delivery.</li>
+    </ul>
+    <h4>When to consider something else</h4>
+    <ul class="usa-content-list">
+      <li><strong>Documents are optional.</strong> Don’t ask users to provide documents if you do not require them.</li>
+      <li><strong>Upload functionality.</strong> The USWDS file input does not currently upload files.</li>
```

Comment:
This might need some clarification. I think the point we're making is that it doesn't _immediately_ upload when the user selects the file(s.) Or are we saying something different? Perhaps mention what it does do ("attaches files which will be uploaded upon form submission," or something like that.) Why would someone need upload functionality, though?

Reply:
Maybe another "when to consider something else" case is when the user is uploading very large files.

Reply:
I've changed this to say that the file input only posts files when the form is submitted. The bold text says "upload in place"

I think regarding the file size, we can say something like "asking for large files" so that we respect data limits or spotty signals.